LucidGloves/src/ControllerDriver.cpp

```diff
@@ -11,11 +11,16 @@ ControllerDriver::ControllerDriver(const VRDeviceConfiguration_t &configuration)
 	);
 
 
-	switch (m_configuration.protocol) {
-	case VRDeviceProtocol::SERIAL:
+	switch (m_configuration.communicationProtocol) {
+	case VRCommunicationProtocol::SERIAL:
 		m_communicationManager = std::make_unique<SerialManager>(m_configuration.serialConfiguration);
 		break;
 	}
+	switch (m_configuration.encodingProtocol) {
```

Comment:
We have the same problem in the serial switch statement as well, but this could cause problems if the value in the configuration doesn't match any cases.

```
switch (m_configuration.communicationProtocol) {
		default:
			DriverLog("No communication protocol defined... using default.");
		case VRCommunicationProtocol::SERIAL:
			m_communicationManager = std::make_unique<SerialManager>(m_configuration.serialConfiguration);
			break;
	}
	switch (m_configuration.encodingProtocol) {
		default:
			DriverLog("No encoding protocol defined... using default.");
		case VREncodingProtocol::LEGACY:
			m_encodingManager = std::make_unique<LegacyEncodingManager>((float)m_configuration.adcCounts);
			break;
	}
```
Something like this should work

Reply:
Ah, yeah I ran into that issue when I had a malformed json. Adding in a default is a good idea